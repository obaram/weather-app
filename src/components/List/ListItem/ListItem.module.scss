@keyframes fadeIn {
    0% {opacity: 0;}
    100% {opacity: 1;}
 }

.wrapper{
    animation: fadeIn ease 0.5s;
    width:100%;
}

.wrapper td{
    vertical-align:middle;
}

.icon{
    margin-right:5px;
}

.button{
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
}