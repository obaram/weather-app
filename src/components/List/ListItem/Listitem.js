import React from 'react';
import { Button } from 'react-bootstrap';
import {Link} from 'react-router-dom';
import AppContext from '../../../Context'
import styles from './ListItem.module.scss'
import FontAwesome from 'react-fontawesome'

const Listitem = ({index,item,unit,handleDelete}) => (
  

  <AppContext.Consumer>
  { (context) => (

  <tr className={styles.wrapper}>
    <td>{index+1}</td>
    <td><Link to={{pathname: `/itemview/${item.city}`, query: { item:item}}}>{item.city}</Link></td>
    <td>{item.temp} {unit==="metric" ? <span>°C</span> : <span>°F</span>}</td>
    <td><Button variant="outline-danger" onClick = {(e) => handleDelete(e,index)} className={styles.button}>

    <FontAwesome
        name='remove'
        size='1px'
        className={styles.icon}
      />  
      
      
      
     Delete</Button></td>
 </tr>


  )}
  </AppContext.Consumer>


)

export {Listitem};