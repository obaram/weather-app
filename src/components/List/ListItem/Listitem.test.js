import React from 'react';
import ReactDOM from 'react-dom';
import {Listitem} from './ListItem';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'; 
import {BrowserRouter,Route, Switch} from 'react-router-dom'
import "../../../../setupTest"


     const props = {
        index: 1,
        item: {
            city: "RzeszÃ³w",
            temp: 25.00,
            lat: 31.165,
            lon:  21.276,
        },
        handleDelete: jest.fn(),   
    } 

  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><Listitem {...props} /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });


 describe('ListItem', () => {
     it('should render list item',()=>{
     
     const wrapper = shallow(<Listitem {...props} />);
     expect(wrapper).toMatchSnapshot();

     })

})
