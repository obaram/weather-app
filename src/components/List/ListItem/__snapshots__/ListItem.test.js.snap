// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render list item 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`renders correctly 1`] = `
<tr
  className="wrapper"
>
  <td>
    2
  </td>
  <td>
    <a
      href="/itemview/Rzeszów"
      onClick={[Function]}
    >
      Rzeszów
    </a>
  </td>
  <td>
    25
     
    <span>
      °F
    </span>
  </td>
  <td>
    <button
      className="button btn btn-outline-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden={true}
        className="fa fa-remove fa-1px icon"
      />
      Usuń
    </button>
  </td>
</tr>
`;
